js-core-2/lesson7.md

```diff
@@ -1,35 +1,217 @@
 # JavaScript Core 4
-** What we will learn today?**
-ToDo
+## What we will learn today?
+- HTTP
+- JSON
+- AJAX
+---
+
+## HTTP
+
+### What's a server?
+
+A device or program that **provides functionality to other programs or devices**. There are database servers, mail servers, game servers, etc.
+
+They can take the form of industrial server farms that provide a service to millions of users (used by Facebook, Google, etc.), to personal servers for storing your files.
+
+The server communicates with **clients**. Clients can be a web browser, a Slack app, your phone, etc.
+
+Client–server systems use the **request–response model**: a client sends a request to the server, which performs some action and sends a response back to the client, typically with a result or acknowledgement.
+
+
+>An example: We can use the Slack app (the client) to put our messages or pictures on Slack. The content is stored on the Slack servers and other clients can then also access the pictures.
+
+### HTTP requests
+
+A server stores the data, and the client (other programs or computers) requests data or sends some of its own. But how do they talk to each other?
+
+**For the client and the server to communicate they need an established an agreed upon way of communicating (a protocol)**. Which is what HTTP (Hypertext Transfer Protocol) is for. It defines the methods you can use to communicate with a server and indicate your desired actions on the resources of the server.
+
+There are two main types of requests: GET and POST.
+
+>With a **GET request** you can ask for specified resource (e.g. show me that Slack photo).
+
+>With a **POST request** you can send content to the server to be appended to the web resource (e.g. post a photo on Slack).
+
+HTTP is the language of the internet. In our case we're using HTML and Javascript, but you can send HTTP requests with other laguages as well.
+
+### AJAX (= Asynchronous JavaScript And XML)
+
+AJAX is a set of useful methods for implementing client-server communication.
+
+![AJAX Diagram](https://www.w3schools.com/xml/ajax.gif "AJAX Diagram")
+
+AJAX just uses a combination of:
+
+> - A browser built-in XMLHttpRequest object (to request data from a web server)
+> - JavaScript and HTML DOM (to display or use the data)
+
+
+**Ajax works behind the scenes, helping the webpage communicate with the server (with GET and POST requests).**
```

Comment:
You should capitalise AJAX to maintain consistency with other references of the term.